### ISSUE TITLE
Making modules

### ISSUE THREAD

**Issue by majimboo** [pizzapanther/GAE-Bulk-Mailer#3]
Is it possible to make modules to create support for services other than AWS and GAE? like my custom service. Also does it use any database to remember the statistics and who to send the email? Will be testing this out tonight.

**Comment by pizzapanther** [comment]
Yes you can model your mailer after the ones in https://github.com/pizzapanther/GAE-Bulk-Mailer/tree/master/bulkmail/mailers

And then change your mailer setting: https://github.com/pizzapanther/GAE-Bulk-Mailer/blob/master/settings.py#L161

Lastly, it does track stats too.  Here are the models that track the stats.
https://github.com/pizzapanther/GAE-Bulk-Mailer/blob/master/bulkmail/tracking/models.py

This is what the stats look like: http://awesomescreenshot.com/0f62i5wk2c

**Comment by majimboo** [comment]
Great, anyways how does the stats work, just asking for an introduction, does every sent email have a unique hash to remember who it was sent to and who clicked?

**Comment by pizzapanther** [comment]
Yes it looks for links in your email and replaces them with tracking links.  It also adds a tracking pixel to track opens.

**Comment by majimboo** [comment]
Do I just download this repo and proceed to editing the source? There is nothing like `pip install gae-bulk-mailer` to install right? Or where can I find the installation / usage docs.

Thanks for the great app btw.

**Comment by pizzapanther** [comment]
You can't use pip on app engine so checkout: https://github.com/pizzapanther/GAE-Bulk-Mailer/blob/master/INSTALL.md

Since you want to write a custom mailer module, I would suggest forking the repo and cloning that.

**Comment by majimboo** [comment]
And how do I start the web? Just like any django project right? 
I believe there is a GUI, how can I access management?
